nix/process-compose/test.nix

```diff
@@ -1,98 +1,24 @@
-              journalctl -u process-compose.service
-              cat process-compose*log
-            """)
-            raise Exception("Processes not started!")
-          info = self.get_processes(${if config.port == null then "8080" else builtins.toString config.port})
-          print(json.dumps(info))
-          data = { x["name"]: x for x in info["data"] }
-          if ready(data):
-            break
-          else:
-            time.sleep(1)
-        return data
-    
-    process_compose = ProcessCompose()
-  '';
 in
 {
   options = {
-    testScript = mkOption {
-      type = types.nullOr types.str;
-      description = ''
-        If set, add a flake check running nixosTest running this process-compose
-        configuration, followed by the specified testScript.
-
-        Useful if you want to test your configuration in CI.
-
-        The testScript will have access to `process_compose.wait` function that
-        can be used to get the running process information, after waiting for
-        the specified readiness status.
-
-        The check is added only on Linux, inasmuch as nixosTest is not available
-        on Darwin.
-      '';
-      default = null;
-      example = ''
-        process_compose.wait_until(lambda procs:
-          procs["netdata"]["is_ready"] == "Ready"
-        )
-        machine.succeed("curl -v http://localhost:19999/")
-      '';
-    };
     outputs.check = mkOption {
       type = types.nullOr types.package;
       default =
-        if (config.testScript == null || !pkgs.stdenv.isLinux) then null else
-        pkgs.nixosTest {
-          testScript = testLibrary + "\n" + config.testScript;
-          name = "process-compose-${name}-test";
-          nodes.machine = {
-            users.users.tester = {
-              isNormalUser = true;
-            };
-            systemd.services."process-compose-${name}" = {
-              enable = true;
-              wantedBy = [ "default.target" ];
-              serviceConfig = {
-                WorkingDirectory = "/tmp";
-                User = "tester";
-                ExecStart = lib.getExe (pkgs.writeShellApplication {
-                  name = "process-compose-${name}";
-                  text = ''
-                    set -x
-                    ${lib.getExe config.outputs.package} -t=false
-                    echo "unexpected: process-compose exited successfully (all processes are completed?)"
-                    exit 2
-                  '';
-                });
-              };
-            };
-          };
-        };
+        if (builtins.hasAttr "test" config.settings.processes) then
+          pkgs.runCommand "${name}-test" { nativeBuildInputs = [ config.outputs.testPackage ]; } ''
+            # Set pipefail option for safer bash
+            set -euo pipefail
+            export HOME=$TMP
+            cd $HOME
+            # Run with tui disabled because /dev/tty is disabled in the simulated shell
+            ${name} -t=false
```

Comment:
It is unclear what `${name}` refers to here. Maybe best to just use `${lib.getExe config.outputs.testPackage}` directly.